rp2040-hal/src/i2c.rs

```diff
@@ -327,12 +327,32 @@ macro_rules! hal {
                     system_clock: SystemF) -> Self
                 where
                     F: Into<HertzU32>,
-                    Sda: ValidPinSda<$I2CX>,
-                    Scl: ValidPinScl<$I2CX>,
+                    Sda: ValidPinSda<$I2CX> + AnyPin<Pull = PullUp>,
+                    Scl: ValidPinScl<$I2CX> + AnyPin<Pull = PullUp>,
                     SystemF: Into<HertzU32>,
                 {
                     Self::new_controller(i2c, sda_pin, scl_pin, freq.into(), resets, system_clock.into())
                 }
+
+                $crate::paste::paste! {
+                    /// Configures the I2C peripheral to work in master mode
+                    /// This function can be called without the pull-ups on the I2C pins.
+                    pub fn [<$i2cX _unchecked>]<F, SystemF>(
```

Comment:
I'm not convinced by the `_unchecked` suffix, as it's otherwise often used to skip some runtime check. Any better ideas?

Reply:
How about a `new_with_external_pull_up` ?

Reply:
That wouldn't fit. The difference is just that this function doesn't check the pull type. It could still use internal pull-ups.
Perhaps `new_unchecked_pull_mode`?
IMHO for this method it doesn't matter if the name is long and ugly. I don't expect it to be used very often. It's just for completeness, in case someone wants to do something strange.

Reply:
I like `new_with_external_pull_up` a lot better. The meaning of "unchecked" in this context is not very clear. Having external pull-ups is quite common in general and might be desirable to tailor your design to your EMI and speed constraints.

Reply:
Ok, changed the function name. I still omitted the `new` part, because it would be inconsistent. Now the two functions available are:
```
    pub fn i2c0<F, SystemF>(
        i2c: I2C0,
        sda_pin: Sda,
        scl_pin: Scl,
        freq: F,
        resets: &mut RESETS,
        system_clock: SystemF,
    ) -> Self
    where
        F: Into<HertzU32>,
        Sda: ValidPinSda<I2C0> + AnyPin<Pull = PullUp>,
        Scl: ValidPinScl<I2C0> + AnyPin<Pull = PullUp>,
        SystemF: Into<HertzU32>,
```
```
    pub fn i2c0_with_external_pull_up<F, SystemF>(
        i2c: I2C0,
        sda_pin: Sda,
        scl_pin: Scl,
        freq: F,
        resets: &mut RESETS,
        system_clock: SystemF,
    ) -> Self
    where
        F: Into<HertzU32>,
        Sda: ValidPinSda<I2C0>,
        Scl: ValidPinScl<I2C0>,
        SystemF: Into<HertzU32>,
```